copier.go

```diff
@@ -123,43 +123,127 @@ func TestCopier(t *testing.T, store types.Storager) {
 				})
 			})
 		})
-
-		 if d, ok := store.(types.Direr); ok {
-			 Convey("When Copy to an existing dir", func() {
-				 srcSize := rand.Int63n(4 * 1024 * 1024) // Max file size is 4MB
-				 r := io.LimitReader(randbytes.NewRand(), srcSize)
-				 src := uuid.New().String()
-
-				 _, err := store.Write(src, r, srcSize)
-				 if err != nil {
-					 t.Fatal(err)
-				 }
-
-				 defer func() {
-					 err = store.Delete(src)
-					 if err != nil {
-						 t.Error(err)
-					 }
-				 }()
-
-				 dst := uuid.New().String()
-				 _, err = d.CreateDir(dst)
-				 if err != nil {
-					 t.Fatal(err)
-				 }
-
-				 defer func() {
-					 err = store.Delete(dst, pairs.WithObjectMode(types.ModeDir))
-					 if err != nil {
-						 t.Error(err)
-					 }
-				 }()
-
-				 err = c.Copy(src, dst)
-				 Convey("The error should be ErrObjectModeInvalid", func() {
-					 So(errors.Is(err, services.ErrObjectModeInvalid), ShouldBeTrue)
-				 })
-			 })
-		 }
 	})
 }
+
+func TestCopierWithDir(t *testing.T, store types.Storager) {
+	TestCopier(t, store)
+
+	if d, ok := store.(types.Direr); ok {
```

Comment:
The check is not needed, as it's a service duty.

Reply:
Got it.